### ISSUE TITLE
 The prop `maxPropObjectKeys` is marked as required in `PropVal`, but its value is `undefined`

### ISSUE THREAD

**Issue by Stupidism** [storybooks/storybook#1305]
My repo: https://github.com/Stupidism/stupid-rc-starter/tree/master/starter
```
warning.js?8a56:36 Warning: Failed prop type: The prop `maxPropObjectKeys` is marked as required in `PropVal`, but its value is `undefined`.
    in PropVal (created by PropVal)
    in PropVal (created by Props)
    in span (created by Props)
    in span (created by Props)
    in span (created by Props)
    in span (created by Props)
    in Props (created by Node)
    in div (created by Node)
    in div (created by Node)
    in Node (created by Story)
    in pre (created by Pre)
    in Pre (created by Story)
    in div (created by Story)
    in div (created by Story)
    in div (created by Story)
    in div (created by Story)
```
I looked at the code. There's no defaultProps [here](https://github.com/storybooks/storybook/pull/1157/files#diff-1010713fd56c690303cda62b997a7f10R118).

But maybe there should no be because, theoretically, all these props come from [Story](https://github.com/storybooks/storybook/pull/1157/files#diff-105d24332ddd34d7e8dcba2ef7c8f26cR12), which has defaultProps.

---
Found the real location: https://github.com/storybooks/storybook/blob/master/addons/info/src/components/PropVal.js#L64

**Comment by Stupidism** [comment]
Temp fix in config.js:
```
// temp fix PropVal.propTypes
PropVal.propTypes = {
  ...PropVal.propTypes,
  maxPropObjectKeys: PropTypes.number,
  maxPropArrayLength: PropTypes.number,
  maxPropStringLength: PropTypes.number,
};
```

**Comment by tmeasday** [comment]
Hi @Stupidism -- I don't quite understand this. How do I see the issue when running your reproduction?

**Comment by Stupidism** [comment]
clone repo
`yarn install && npm start`

**Comment by tmeasday** [comment]
Ok, thanks I can see a problem, although not the one you reported about. How do I see the `PropVal` issue? (I see an issue for `Props` when browsing to the `RenderCounter`)

It would be great to create a simpler reproduction of the issue--perhaps a variation of the current `cra-kitchen-sink` project here: https://github.com/storybooks/storybook/blob/master/examples/cra-kitchen-sink/src/stories/index.js#L48

**Comment by Stupidism** [comment]
Oh, because I fixed it temporarily by modifying the propTypes in this [commit](https://github.com/Stupidism/stupid-rc-starter/commit/bf954db9a39694e549fb20daa85ff94576937b41).

try `git revert bf954db9a39694e549fb20daa85ff94576937b41` in my repo.

>Temp fix in config.js:

**Comment by Li0liQ** [comment]
The issue is reproduced by passing an array as a prop.
It was introduced in 
https://github.com/storybooks/storybook/commit/70f7b51f615f3b2bcfd7ba26343cde7adbb328ec by adding required props:
```
PropVal.propTypes = {
  val: PropTypes.any.isRequired, // eslint-disable-line
  maxPropObjectKeys: PropTypes.number.isRequired,
  maxPropArrayLength: PropTypes.number.isRequired,
  maxPropStringLength: PropTypes.number.isRequired,
};
```
The lines causing it to fail are:
https://github.com/storybooks/storybook/blob/master/addons/info/src/components/PropVal.js#L43
https://github.com/storybooks/storybook/blob/master/addons/info/src/components/PropVal.js#L64

Basically, in case of array/object elements the only prop being passed is `val`. Hence, validation errors.
